tenant/pkg/controller/tenantnamespace/tenantnamespace_controller.go

```diff
@@ -88,5 +99,50 @@ func (r *ReconcileTenantNamespace) Reconcile(request reconcile.Request) (reconci
 		return reconcile.Result{}, err
 	}
 
+	nsList := &corev1.NamespaceList{}
+	err = r.List(context.TODO(), &client.ListOptions{}, nsList)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	expectedOwnerRef := metav1.OwnerReference{
+		APIVersion: tenancyv1alpha1.SchemeGroupVersion.String(),
+		Kind:       "TenantNamespace",
+		Name:       instance.Name,
+		UID:        instance.UID,
+	}
+	for _, each := range nsList.Items {
+		if each.Name == instance.Spec.Name {
+			// Check OwnerReference
+			found := false
+			for _, ownerRef := range each.OwnerReferences {
+				if ownerRef == expectedOwnerRef {
+					found = true
+					break
+				} else if ownerRef.APIVersion == expectedOwnerRef.APIVersion && ownerRef.Kind == expectedOwnerRef.Kind {
+					// The namespace is owned by another TenantNamespace CR, fail the reconcile
+					err = fmt.Errorf("Namespace %v is owned by another %v TenantNamespace CR", each.Name, ownerRef)
+					return reconcile.Result{}, err
+				}
+			}
+			if !found {
+				log.Info("Namespace has been created without TenantNamespace owner", "namespace", each.Name)
+				// TODO: grab the namespace ownership
+			}
+			return reconcile.Result{}, nil
```

Comment:
Exiting early here will probably need to change once we start doing other things inside the namespace that's been created

Reply:
Yes. This is going to be a small change though. I'd like to keep each change as concise as possible.

Reply:
> I expected the name of the Namespace created to match the name of the TenantNamespace. This would prevent (at creation time) any name conflicts that might arise from tenants already claiming the name.

A TenantNamespace will be namespace scoped while namespaces are cluster scoped. So you'd still have possible collisions. However, I tend to agree that that TenantNamespace name would map to the namespace name.